webui/react/src/e2e/tests/projects.spec.ts

```diff
@@ -255,4 +255,73 @@ test.describe('Project List', () => {
     await firstRow.pwLocator.waitFor({ state: 'hidden' });
     await firstCard.pwLocator.waitFor();
   });
+
+  test.describe('List View', () => {
+    let newProject: V1PostProjectResponse;
+    let destinationWorkspace: V1Workspace;
+
+    test.beforeAll(async ({ backgroundApiProject, backgroundApiWorkspace }) => {
+      newProject = await backgroundApiProject.createProject(
+        workspaceIds[0],
+        backgroundApiProject.new(),
+      );
+      destinationWorkspace = (
+        await backgroundApiWorkspace.createWorkspace(backgroundApiWorkspace.new())
+      ).workspace;
+    });
+    test.beforeEach(async ({ authedPage }) => {
+      const workspaceDetails = new WorkspaceDetails(authedPage);
+
+      const workspaceProjects = workspaceDetails.workspaceProjects;
+
+      await workspaceProjects.gridListRadioGroup.list.pwLocator.click();
+    });
+    test.afterAll(async ({ backgroundApiWorkspace }) => {
+      await backgroundApiWorkspace.deleteWorkspace(destinationWorkspace.id);
+    });
+
+    test('move a project to a different workspace', async ({ authedPage }) => {
+      // eslint-disable-next-line no-console
+      console.log(newProject.project.name);
+      const workspaceDetails = new WorkspaceDetails(authedPage);
+      const workspaceProjects = workspaceDetails.workspaceProjects;
+
+      const projectsTable = workspaceProjects.table.table;
+
+      let newProjectRow = (
+        await projectsTable.filterRows(
+          async (row) => (await row.name.pwLocator.textContent()) === newProject.project.name,
+        )
+      )[0];
+
+      expect(await newProjectRow.name.pwLocator.innerText()).toBe(newProject.project.name);
+
+      const moveMenuItem = newProjectRow.action;
+
+      await moveMenuItem.open();
+      await moveMenuItem.move.pwLocator.click();
+      await workspaceProjects.moveModal.destinationWorkspace.pwLocator.fill(
+        destinationWorkspace.name,
+      );
+      await workspaceProjects.moveModal.destinationWorkspace.pwLocator.press('Enter');
+      await workspaceProjects.moveModal.footer.submit.pwLocator.click();
+
+      await workspaceProjects.moveModal.pwLocator.waitFor({ state: 'hidden' });
+      await newProjectRow.pwLocator.waitFor({ state: 'hidden' });
+
+      newProjectRow = (
+        await projectsTable.filterRows(
+          async (row) => (await row.name.pwLocator.textContent()) === newProject.project.name,
+        )
+      )[0];
+
+      await expect(newProjectRow).toBeUndefined();
```

Comment:
i think expect length 0 might be more clear, i also think line 310 is good enough here